src/test/java/htsjdk/samtools/fastq/FastqEncoderTest.java

```diff
@@ -65,12 +67,30 @@ public void testAsSAMRecord() throws Exception {
         testConvertedSAMRecord(FastqEncoder.asSAMRecord(fastqRecord, samRecord.getHeader()), samRecord);
         fastqRecord = new FastqRecord(samRecord.getReadName() + FastqConstants.SECOND_OF_PAIR, samRecord.getReadBases(), "", samRecord.getBaseQualities());
         testConvertedSAMRecord(FastqEncoder.asSAMRecord(fastqRecord, samRecord.getHeader()), samRecord);
+        fastqRecord = new FastqRecord(samRecord.getReadName() + FastqConstants.SECOND_OF_PAIR, samRecord.getReadBases(), "Quality header comment", samRecord.getBaseQualities());
+        // default method does not include the comment header
+        testConvertedSAMRecord(FastqEncoder.asSAMRecord(fastqRecord, samRecord.getHeader()), samRecord);
+        // test with qualityHeaderToComment=true populates the CO tag
+        samRecord.setAttribute("CO", fastqRecord.getBaseQualityHeader());
+        testConvertedSAMRecord(FastqEncoder.asSAMRecord(fastqRecord, samRecord.getHeader(), FastqEncoder.QUALITY_HEADER_TO_COMMENT_TAG), samRecord);
+    }
+
+    @Test
+    public void testAsSAMRecordParsedTags() throws Exception {
+        final String tags = "BC:Z:ACTG\tRG:Z:sample1";
+        final SAMRecord samRecord = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, "10M", null, 2);
+        final FastqRecord record = new FastqRecord(samRecord.getReadName(), samRecord.getReadBases(), tags, samRecord.getBaseQualities());
+        final SAMRecord converted = FastqEncoder.asSAMRecord(record, samRecord.getHeader(), FastqEncoder.QUALITY_HEADER_PARSE_SAM_TAGS);
+        testConvertedSAMRecord(converted, samRecord);
+        Assert.assertEquals(converted.getAttribute("BC"), "ACTG");
```

Comment:
use SAMTag variable here as well. if they do not exist, then please add them.

Reply:
Done